File: Crio/crio_programming_interview_problems-master/TreePathSum/TreePathSum.cpp
#include <bits/stdc++.h>
#include "Solution.cpp"
using namespace std;

// CRIO_SOLUTION_START_MODULE_TREE_PATH_SUM
// CRIO_SOLUTION_END_MODULE_TREE_PATH_SUM

// CRIO_SOLUTION_START_MODULE_TREE_PATH_SUM_WITH_BREAKDOWN
void test()
{
    assert(true);
}
// CRIO_SOLUTION_END_MODULE_TREE_PATH_SUM_WITH_BREAKDOWN

int main()
{
// CRIO_SOLUTION_START_MODULE_TREE_PATH_SUM_WITH_BREAKDOWN
    test();
// CRIO_SOLUTION_END_MODULE_TREE_PATH_SUM_WITH_BREAKDOWN

    TreeNode* root  = TreeNode().readTreeReturnRoot();
    int k;
    cin >> k;
    printf("%d\n", CountTreePathSumToK(root, k));
    return 0;
}

File: Crio/crio_programming_interview_problems-master/TreePathSum/TreePathSum.py
from crio.ds.Tree.TreeNode import *
from Solution import countTreePathsThatSumsToK

# CRIO_SOLUTION_START_MODULE_TREE_PATH_SUM
# CRIO_SOLUTION_END_MODULE_TREE_PATH_SUM


# CRIO_SOLUTION_START_MODULE_TREE_PATH_SUM_WITH_BREAKDOWN
def test():
    assert True
# CRIO_SOLUTION_END_MODULE_TREE_PATH_SUM_WITH_BREAKDOWN


def main():

# CRIO_SOLUTION_START_MODULE_TREE_PATH_SUM_WITH_BREAKDOWN
    test()
# CRIO_SOLUTION_END_MODULE_TREE_PATH_SUM_WITH_BREAKDOWN

    root = readTreeReturnRoot()
    k = int(input())
    print(countTreePathsThatSumsToK(root, k))

if __name__ == '__main__':
    import sys
    sys.setrecursionlimit(100000)
    main()
